src/db/spanner/manager.rs

```diff
@@ -70,24 +88,17 @@ impl ManageConnection for SpannerConnectionManager {
         Ok(SpannerSession {
             client,
             session,
-            use_test_transactions: false,
+            use_test_transactions: self.test_transactions,
         })
     }
 
-    fn is_valid(&self, conn: &mut Self::Connection) -> Result<(), Self::Error> {
+    async fn is_valid(&self, conn: Self::Connection) -> Result<Self::Connection, Self::Error> {
         let mut req = GetSessionRequest::new();
         req.set_name(conn.session.get_name().to_owned());
         if let Err(e) = conn.client.get_session(&req) {
```

Comment:
Replace `get_session` with `get_session_async`.

Reply:
This got reverted back to `get_session`, needs a re-revert or something back to `get_session_async`.